gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -542,17 +560,32 @@ public void start(Listener<RespT> responseListener, Metadata headers) {
             new SimpleForwardingClientCallListener<RespT>(responseListener) {
               @Override
               public void onClose(Status status, Metadata trailers) {
-                try {
+                if (wasClosed.compareAndSet(false, true)) {
                   super.onClose(status, trailers);
-                } finally {
-                  entry.release();
+                  if (wasReleased.compareAndSet(false, true)) {
+                    entry.release();
+                  } else {
+                    LOG.log(
+                        Level.WARNING,
+                        "Call was not closed but entry was released. This may be due to an exception on start of the call.");
```

Comment:
I would reword this to "channel refcount was released before the rpc closed. This may be due to ..."

Reply:
Because `channel refcount` is easier to understand than `entry`? Or put more focus on `released before closed`?